### DISCUSSION TITLE
Calibration full cycle damage vs max charge damage

### DISCUSSION THREAD

**Discussion by joranr** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#327] [Q&A]
in the wiki I read the following:

IMPORTANT: Keeping your battery at a lower percentage, such as 20-70%, over weeks without doing full cycles (100%-0%) can result in a disturbed battery calibration. When this happens, your Macbook might turn off with 40-50% left or your battery capacity will drop significantly. However, this is only due to a disturbed battery calibration and not because of a faulty or degraded battery. To avoid this issue, we recommend doing at least one full cycle (0%-100%) every month. Even if your battery calibration gets disturbed, doing 4-5 full cycles will recalibrate your battery and the capacity will go up again.

I've also read online that doing a full discharge is very bad for the battery. Will the benefit of not keeping the battery fully charged when on the adapter outweigh the damage that will be done when fully discharging the battery every month?

Also, what is the optimal battery percentage for longevity if I keep my M1 air on the charger for a long time? 40%, 50% or higher?

Thanks for reading

**Comment by MatthiasKerbl** [comment]
Hi @joranr,

Apples says that modern MacBook batteries should last up to 1000 cycles with about 80% capacity left. Therefore, doing one full cycle every two weeks shouldn’t damage the battery significantly. Keeping your battery plugged in most of the time at 100%, on the other hand, will result in much faster battery degradation.
It's difficult to say what the best percentage for longevity is. This is due to batteries changing and improving all the time. From what I read, keeping your battery at 80% should be good enough but I also saw some research indicating that as low as 40% could be even better. However, keeping your battery at 40% for a long time will mess up your battery calibration much faster.

**Reply by joranr** [reply]
Thanks for the reply,

from what I understand
5 * charging from 50% to 70% equals 1 cycle but letting a battery go to 0% and then charge to 100% should be worse for the battery (https://batteryuniversity.com/article/bu-808-how-to-prolong-lithium-based-batteries) so I don't think cycles is an accurate measure.

Do you have a source on the fact that keeping your battery plugged in most of the time at 100%? 

As for the fact of 80% or 40% the same url posted above contains a table of 40% vs 100% SOC capacity loss percentages. I assume that a 40% charge will increase longevity opposed to 80%. Do you have any sources on how fast calibration issues emerge? If I put al dente on 40% will apple show a lower or higher percentage battery charge after a while? Or won't it charge to a 100% when I remove the charge limit after a long time?

Thanks

**Comment by joranr** [comment]
Can you please show me the source for this

"IMPORTANT: Keeping your battery at a lower percentage, such as 20-80%, over weeks without doing full cycles (100%-0%) can result in a disturbed battery calibration. When this happens, your Macbook might turn off with 40-50% left or your battery capacity will drop significantly. However, this is only due to a disturbed battery calibration and not because of a faulty or degraded battery. To avoid this issue, we recommend doing at least one full cycle (0%-100%) every two weeks. Even if your battery calibration gets disturbed, doing 4+ full cycles will recalibrate your battery and the capacity will go up again."

and this

"Keeping your battery at a lower percentage, such as 20-70%, over weeks without doing full cycles (0%-100%-0%) can result in a disturbed battery calibration. When this happens, your Macbook might turn off with 40-50% left or your battery capacity will drop significantly. However, this is only due to a disturbed battery calibration and not because of a faulty or degraded battery. To avoid this issue, we recommend doing at least one full cycle (0%-100%) every month. Even if your battery calibration gets disturbed, doing 4-5 full cycles will recalibrate your battery and the capacity will go up again."

statement?

I've read completely discharing a battery to 0% is very bad for a battery. 
Also, why wouldn't keeping it at 80% mess up the calibration, while keeping it below 80% or 70% will ?

**Reply by mikeatlarge** [reply]
There are many references available (just use your favorite search engine) on Li-Ion battery longevity when keeping batteries fully charged. Dell builds this feature into their BIOS on most of their laptops, for example, so you can set the maximum charge level or have it be adaptive based on your usage. Electric and hybrid cars do the same thing. There's no debate about the harm from keeping the battery at 100% charge. Even Apple finally acknowledged this issue in their latest MacBooks, with the "Optimize Charging" feature, they just didn't implement it very well as it still often charges to 100% when it doesn't need to.

The battery management circuitry prevents over discharging the MacBook (or most any laptop) battery. There is no significant harm being done to occasionally fully discharge your MacBook as the circuitry protects the battery. The danger is having it fully discharged and storing it that way as the battery will very slowly discharge further which could result in eventual harm. But if you charge it promptly when it's fully discharged, there's no issue.

The only issue here is how MacBooks calculate the charge level and remaining run time. It's much better for your MacBook battery to use AlDente rather than not use it. As the developer has said, it cannot harm the battery only alter the calibration which can be corrected.

The calibration looks at the run time over several charge cycles. If the algorithm sees less overall run time because it's starting out at say 80% instead of 100% it might assumes the battery has lost around 20% of its capacity. On the new M1 MacBooks this may be less of an issue as they were designed for adaptive charging. Apple doesn't disclose such things so we can only guess how their battery management works and its safe to assume it's different over various generations and models of MacBooks. For example my 2015 MacBook Air and Pro don't even offer the "Optimize Battery Charging" feature in BigSur like my M1 MacBook does. 

If you're still concerned buy the Pro license which has additional features for managing this issue.
